subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/AbstractJvmLibraryInitIntegrationSpec.groovy

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.buildinit.plugins
+
+import org.gradle.integtests.fixtures.executer.ExecutionResult
+
+abstract class AbstractJvmLibraryInitIntegrationSpec extends AbstractInitIntegrationSpec{
+
+    @Override
+    String subprojectName() { 'lib' }
+
+    @Override
+    protected ExecutionResult run(String... tasks) {
```

Comment:
I don't think we should override what "run" means.  This is going to be a foot-gun in the future where run/succeeds/fails all behave differently WRT toolchains.

I think if you add a `setup` method, you can do `executer.withToolchainDetectionEnabled()` to have it always enabled or we can only enable this on tests that really require it.